### PR TITLE
Add typing.TypeVarTuple, Unpack, assert_type

### PR DESCRIPTION
**Pull request by JelleZijlstra**
python/cpython#31021 and python/cpython#30843.

### CODE REVIEW

File: stdlib/typing.pyi [hunk 1]
```diff
@@ -524,6 +527,12 @@ if sys.version_info >= (3, 8):
 if sys.version_info >= (3, 11):
     Self: _SpecialForm
     Never: _SpecialForm = ...
+    Unpack: _SpecialForm
+
+    class TypeVarTuple:
+        __name__: str
```
Comment (JelleZijlstra) [inline stdlib/typing.pyi:533]: Anticipating python/cpython#31954.

Comment (AlexWaygood) [reply stdlib/typing.pyi:533]: Should we wait until that's merged before merging this, or do you reckon we're good to go?

Comment (JelleZijlstra) [reply stdlib/typing.pyi:533]: Better to wait I think.

Comment (AlexWaygood) [reply stdlib/typing.pyi:533]: (Converting this to a draft for now so it doesn't get accidentally merged in the meantime)

Comment (JelleZijlstra) [reply stdlib/typing.pyi:533]: I merged the CPython change

Comment (AlexWaygood) [reply stdlib/typing.pyi:533]: And I merged this one :)

**Pull request merged**